modules/apps/transfer/keeper/keeper.go

```diff
@@ -133,14 +137,83 @@ func (k Keeper) IterateDenomTraces(ctx sdk.Context, cb func(denomTrace types.Den
 
 	defer sdk.LogDeferred(ctx.Logger(), func() error { return iterator.Close() })
 	for ; iterator.Valid(); iterator.Next() {
-
 		denomTrace := k.MustUnmarshalDenomTrace(iterator.Value())
 		if cb(denomTrace) {
 			break
 		}
 	}
 }
 
+// GetTotalEscrowForDenom gets the total amount of source chain tokens that are in escrow.
+func (k Keeper) GetTotalEscrowForDenom(ctx sdk.Context, denom string) math.Int {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.TotalEscrowForDenomKey(denom))
+	if bz == nil {
+		return math.ZeroInt()
+	}
+
+	var amount math.Int
+	if err := amount.Unmarshal(bz); err != nil {
+		panic(err)
+	}
+	return amount
+}
+
+// SetTotalEscrowForDenom stores the total amount of source chain tokens that are in escrow.
+func (k Keeper) SetTotalEscrowForDenom(ctx sdk.Context, denom string, amount math.Int) {
+	if amount.LT(math.ZeroInt()) {
+		panic(fmt.Sprintf("amount cannot be negative: %s", amount))
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	bz, err := amount.Marshal()
+	if err != nil {
+		panic(err)
+	}
+
+	store.Set(types.TotalEscrowForDenomKey(denom), bz)
+}
+
+// GetAllDenomEscrows returns the escrow information for all the denominations.
+func (k Keeper) GetAllDenomEscrows(ctx sdk.Context) sdk.Coins {
+	escrows := sdk.Coins{}
+	k.IterateDenomEscrows(ctx, func(denomEscrow sdk.Coin) bool {
+		escrows = append(escrows, denomEscrow)
+		return false
+	})
+
+	return escrows.Sort()
+}
+
+// IterateDenomEscrows iterates over the denomination escrows in the store
+// and performs a callback function.
+func (k Keeper) IterateDenomEscrows(ctx sdk.Context, cb func(denomEscrow sdk.Coin) bool) {
```

Comment:
A new convention we want to do is have every iteration function take in a [`prefix`](https://github.com/cosmos/ibc-go/issues/2867), this allows the caller to decide the iteration starting point without creating a secondary function